backend/src/middlewares/rentalRequests.ts

```diff
@@ -0,0 +1,106 @@
+import { IRequest } from '../routes';
+import { NextFunction, Response } from 'express';
+import { Types } from 'mongoose';
+import Publication from '../models/publication';
+import { getPublicationByKey } from '../services/publication';
+import { isPublicationOwner } from '../routes/permissions/publications';
+import {
+    canDeleteRentalRequest,
+    canGetRentalRequests,
+} from '../routes/permissions/rentalRequests';
+import RentalRequest from '../models/rentalRequest';
+
+const authCreateRentalRequest = async (
+    req: IRequest,
+    res: Response,
+    next: NextFunction,
+) => {
+    const { publicationId } = req.body;
+
+    try {
+        const publication = await Publication.findOne({
+            _id: Types.ObjectId(publicationId),
+        });
+
+        if (!publication) {
+            return res
+                .status(400)
+                .json({ msg: 'The publication does not exist' });
+        }
+        if (!publication.status) {
+            return res.status(400).json({
+                msg: 'Cant create request - the publication is not active',
+                publication,
+            });
+        }
+        return next();
+    } catch (error) {
+        return res.status(400).json({ msg: 'Something went wrong', error });
+    }
+};
+
+const authGetRentalRequestsByPublicationId = async (
+    req: IRequest,
+    res: Response,
+    next: NextFunction,
+) => {
+    const { publicationId } = req.params;
+
+    try {
+        const publication = await getPublicationByKey(
+            '_id',
+            Types.ObjectId(publicationId),
+        );
+
+        if (!publication) {
+            return res
+                .status(400)
+                .json({ msg: 'The publication does not exist' });
+        }
+
+        if (!(await canGetRentalRequests(req.user, publication))) {
+            return res.status(400).json({
+                msg: 'Cant get rental requests for this publication - not the owner',
+                publication,
+            });
+        }
+        return next();
+    } catch (error) {
+        return res.status(400).json({ msg: 'Something went wrong', error });
+    }
+};
+
+const authDeleteRentalRequest = async (
+    req: IRequest,
+    res: Response,
+    next: NextFunction,
+) => {
+    const { id } = req.params;
+
+    try {
+        const rentalRequest = await RentalRequest.findOne({
+            _id: Types.ObjectId(id),
+        });
+
+        if (!rentalRequest) {
+            return res
+                .status(404)
+                .json({ msg: 'The rental request does not exist' });
+        }
+
+        if (await canDeleteRentalRequest(req.user, rentalRequest)) {
```

Comment:
```suggestion
        if (!(await canDeleteRentalRequest(req.user, rentalRequest))) {
```
Falto negar la condición